base/math.jl

```diff
@@ -70,11 +71,23 @@ end
 
 # evaluate p[1] + x * (p[2] + x * (....)), i.e. a polynomial via Horner's rule
 macro horner(x, p...)
+    @gensym val
     ex = esc(p[end])
     for i = length(p)-1:-1:1
-        ex = :(muladd(t, $ex, $(esc(p[i]))))
+        ex = :(muladd($val, $ex, $(esc(p[i]))))
     end
-    Expr(:block, :(t = $(esc(x))), ex)
+    Expr(:block, :($val = $(esc(x))), ex)
+end
+
+# evaluate p[1] + x * (p[2] + x * (....)), i.e. a polynomial via Horner's rule
+# and convert coefficients to same type as x
+macro horner_oftype(x, p...)
```

Comment:
Why is this macro needed?  If you change the floating-point type, you generally will change the degree of the polynomial as well, and you can just use literals of the correct type, no?  e.g. it looks like you only use it for `Float64` and `Float32` below, and Julia has a compact syntax for `Float32` literals (append `f0`).